nx/lib/nx.ex

```diff
@@ -7353,7 +7353,7 @@ defmodule Nx do
   def map(tensor, opts \\ [], fun) do
     %T{type: type} = tensor = to_tensor(tensor)
     opts = keyword!(opts, type: type)
-    output_type = opts[:type]
```

Comment:
Did you take a look at things like `case tensor.type do` as well?

Reply:
`tensor.type` should always be in the canonical form, because we always normalize in the creation functions, no?

Reply:
I've done a pass through this file and it seems everything is ok, because all tensor.type occurrences happen after a `to_tensor` call :)

Reply:
To be clear, my point was that we should never encounter `%Nx.Tensor{type: :f32}` (which is reflected in the typespec), if we do it's a bug elsewhere. As far as I understand we only want the short syntax to abbreviate options :)

Reply:
Correct!